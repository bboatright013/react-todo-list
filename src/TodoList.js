import React , { useState } from 'react';
import { v4 as uuid } from 'uuid';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

const TodoList = () => {
    const [todoList, setTodoList] = useState([]);

    const addTodo = todo => {
        let newTodo = {...todo, id: uuid()}
        setTodoList(todos => [...todos, newTodo]);
    }

    const dltTodo = (todoId) => {
        const idxToDlt = todoList.findIndex(todo => todo.id === todoId);
        setTodoList(todos => [...todos.slice(0,idxToDlt), ...todos.slice(idxToDlt + 1)])
    }

    return (
        <div className="TodoList">
            <NewTodoForm addTodo={addTodo}/>
            <ul className="todos">
                {todoList.map(todo => {
                    return <Todo dltTodo={dltTodo} id={todo.id} key={todo.id} task={todo.task} />
                })}
            </ul>
        </div>
    )
}

export default TodoList;