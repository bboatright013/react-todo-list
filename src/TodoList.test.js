import React from 'react';
import {render, screen, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';


//Smoke Test
it("SMOKE: renders without crashing", () => {
    render(<TodoList />);
  });
  
//Snapshot test
it("SNAP: matches snapshot", () => {
    const {asFragment} = render(<TodoList />);
    expect(asFragment).toMatchSnapshot();
  });

it("creates a todo", function() {
    const {container } = render( <TodoList />);

    const button = container.querySelector("button");
    let task = screen.getByPlaceholderText('task');


    task.nodeValue="My new todo";


    fireEvent.click(button);

    const newTodo = container.querySelector("li.Todo");
    expect(newTodo).toBeInTheDocument()
    const dltButton = container.querySelector("span.DeleteButton");

    fireEvent.click(dltButton);

    expect(newTodo).not.toBeInTheDocument();

})

