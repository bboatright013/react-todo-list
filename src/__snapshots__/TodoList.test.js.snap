// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAP: matches snapshot 1`] = `[Function]`;
